### ISSUE TITLE
v1.0-beta initializer_list issues

### ISSUE THREAD

**Issue by voivoid** [ericniebler/range-v3#1004]
Hi!

The following code fails to compile using ranges v3 v1.0-beta branch

```
#include "range/v3/algorithm/min.hpp"

int main() {
    auto xs = { 1, 2, 3 };
    ranges::min( xs ); // compilation fails on gcc 8.2.1, clang 7.0.1

    const auto ys = { 1, 2, 3 };
    ranges::min( ys ); // compilation fails on clang 7.0.1 ( gcc is fine )

    return 0;
}
```
<br/>
Is it ok to make issues concerning beta branch? I'll submit a few more if you think it's appropriate

Thanks!

**Comment by CaseyCarter** [comment]
> Is it ok to make issues concerning beta branch?

Yes, of course - that's most of the purpose of having a beta.

**Comment by CaseyCarter** [comment]
Root cause: In `ranges::min_fn`, the `initializer_list` overload is trying to delegate to an iterator/sentinel overload that doesn't exist and actually calling the `const T&` overload that returns the larger of the `begin` and `end` pointers (hint, it's the `end` pointer) to the `initializer_list` elements.

**Comment by CaseyCarter** [comment]
potential fixes:
* Use the "overloading `initializer_list` with `Range`" idiom:
  ```diff
   template<typename T, typename C = less, typename P = identity>
  -constexpr /*c++14*/ auto operator()(std::initializer_list<T> rng, C pred = C{},
  +constexpr /*c++14*/ auto operator()(std::initializer_list<T> const &&rng, C pred = C{},
           P proj = P{}) const ->
       CPP_ret(T)(
           requires Copyable<T> && IndirectStrictWeakOrder<C, projected<T const *, P>>)
   {
  -    return (*this)(rng.begin(), rng.end(), std::move(pred), std::move(proj));
  +    return (*this)(rng, std::move(pred), std::move(proj));
   }
```
* Factor the body of the `Range` overload out into an internal implementation detail, and call that function from both the `Range` and `initializer_list` overloads.

**Comment by CaseyCarter** [comment]
Closed by #1006.
